rust/lance-index/src/scalar.rs

```diff
@@ -108,7 +108,11 @@ pub trait IndexReader: Send + Sync {
     /// Read the n-th record batch from the file
     async fn read_record_batch(&self, n: u32) -> Result<RecordBatch>;
     /// Read the range of rows from the file
-    async fn read_range(&self, range: std::ops::Range<usize>) -> Result<RecordBatch>;
+    async fn read_range(
+        &self,
+        range: std::ops::Range<usize>,
+        projection: Option<&[&str]>,
+    ) -> Result<RecordBatch>;
```

Comment:
This change seems fine.  Do you want to just briefly document whether this is expected to handle nested references (e.g. `read_range(0..30, Some(&["x.y"]))`)?

Probably easiest if it doesn't for now and we can change it in the future.

Reply:
just added document to say nested column not supported